src/printer.js

```diff
@@ -173,26 +263,92 @@ function printMemberChain(path, options, print) {
   //     ->b->c()
   //     ->d()
   const groups = [];
-  let currentGroup = [];
-  let hasSeenCall = false;
-  printedNodes.forEach(printedNode => {
-    if (hasSeenCall && printedNode.node.kind === "propertylookup") {
-      groups.push(currentGroup);
-      currentGroup = [printedNode];
-      hasSeenCall = false;
+  let currentGroup = [printedNodes[0]];
+  let i = 1;
+  for (; i < printedNodes.length; ++i) {
```

Comment:
what was wrong with foreach? (you could still just use the index from the forEach)

Reply:
@mgrip forEach don't have `break`, need use `for` for this case.